main/src/ca/uwaterloo/flix/language/phase/Monomorpher.scala

```diff
@@ -812,6 +812,12 @@ object Monomorpher {
 
   }
 
+  private def combineEnvs(envs: List[Map[Symbol.VarSym, Symbol.VarSym]]): Map[Symbol.VarSym, Symbol.VarSym] = {
```

Comment:
Cases can have zero terms, so `envs.reduce(_ ++ _)` crashes sometimes because of an empty list